bin/node/runtime/src/lib.rs

```diff
@@ -1343,6 +1345,18 @@ impl_runtime_apis! {
 			Contracts::bare_call(origin, dest, value, gas_limit, input_data)
 		}
 
+		fn instantiate(
+			origin: AccountId,
+			endowment: Balance,
+			gas_limit: u64,
+			code: pallet_contracts_primitives::Code<Hash>,
```

Comment:
What's stopping us from having the same API for the dispatchables instead of the separate `instantiate` and `instantiate_with_code` methods - is it to do with the weight calculation?

Reply:
I think we could match on the `Code` in the weight annotation and through that use the same API. However, the reason for having two separate dispatchables was so that we could leave `instantiate` as is and just add another one. That way we did not break backwards compatibility.

The code inside the dispatchables should be deduplicated eventually.